datajoint/expression.py

```diff
@@ -112,24 +121,69 @@ def from_clause(self):
             )
         return clause
 
-    def where_clause(self):
+    def where_clause(self, restriction_list=None):
         return (
-            ""
-            if not self.restriction
+            " WHERE (%s)" % ")AND(".join(str(s) for s in restriction_list)
+            if restriction_list
             else " WHERE (%s)" % ")AND(".join(str(s) for s in self.restriction)
+            if self.restriction
+            else ""
+        )
+
+    def sorting_clauses(self, limit=None, offset=None, order_by=None):
+        if offset and limit is None:
+            raise DataJointError("limit is required when offset is set")
+        clause = ""
+        if order_by is not None:
+            clause += " ORDER BY " + ", ".join(order_by)
+        if limit is not None:
+            clause += " LIMIT %d" % limit + (" OFFSET %d" % offset if offset else "")
+        return clause
+
+    def make_top_subquery(self, tops, fields=None, i=0):
```

Comment:
I would like why `make_top_subquery` necessary in addition to `make_subquery`. Why is not enough to use a regular subquery?